src/main/java/hyperledger/besu/java/rest/client/model/abi/AbiDefinitionWrapper.java

```diff
@@ -28,30 +28,26 @@ public List<Type> getInputType(Object[] params) {
     if (CollectionUtils.isEmpty(inputs)) {
       return Collections.emptyList();
     }
-    List<Type> inputTypes = new ArrayList<>();
     List<Class> inputClasses =
         inputs.stream()
             .map(AbiDefinition.NamedType::getType)
             .map(AbiTypes::getType)
             .collect(Collectors.toList());
+    if (inputClasses.size() != params.length) {
+      throw new BesuTransactionException(
+          HYPERLEDGER_BESU_NO_ABI_DEFINITION_FOUND_ERROR,
+          "No of input params doesn't match number of inputs for the function");
+    }
     int index = 0;
     for (Class clazz : inputClasses) {
-      Type type;
-      try {
-        type = (Type) clazz.getConstructor(params[index].getClass()).newInstance(params[index]);
-        index++;
-        inputTypes.add(type);
-      } catch (InstantiationException
-          | IllegalAccessException
-          | InvocationTargetException
-          | NoSuchMethodException e) {
+      if (!clazz.equals(params[index].getClass())) {
         throw new BesuTransactionException(
             HYPERLEDGER_BESU_NO_ABI_DEFINITION_FOUND_ERROR,
-            "Unable to retrieve input type from ABI Definition file",
-            e);
+            "Input parameter class doesn't match class of corresponding input for the function");
```

Comment:
We are throwing exception here, it needs to be passed back to the caller, so that user gets this in the response.

Reply:
Execution breaks when this exception is thrown, which response are we talking about?

Reply:
<img width="834" alt="Screenshot 2023-03-16 at 1 49 42 PM" src="https://user-images.githubusercontent.com/102647189/225557455-14b8cae7-eacf-4acd-a1ad-7e8fc318ba26.png">
Error messages are getting propagated if request fails